### PR TITLE
Fix gather for uint8 data type (index_select)

### PR DESCRIPTION
**Pull request by DenisVieriu97**

### CODE REVIEW

File: aten/src/ATen/native/mps/OperationUtils.h [hunk 1]
```diff
@@ -64,7 +64,7 @@ class Placeholder {
  public:
   Placeholder() : _placeholder(nullptr), _value(nullptr), _tensor(Tensor()) {}
   Placeholder(MPSGraphTensor* mpsGraphTensor) : _placeholder(mpsGraphTensor), _value(nullptr), _tensor(Tensor()) {}
-  Placeholder(MPSGraphTensor* mpsGraphTensor, const Tensor& self, MPSShape *mpsShape = nullptr, bool gatherTensorData = true);
+  Placeholder(MPSGraphTensor* mpsGraphTensor, const Tensor& self, MPSShape *mpsShape = nullptr, bool gatherTensorData = true, MPSDataType* dataType = nullptr);
```
Comment (razarmehr) [inline aten/src/ATen/native/mps/OperationUtils.h:67]: Why pass by pointer? You could just declare with default value like `MPSDataType dataType = MPSDataTypeInvalid` and in the code check `if (dataType != MPSDataTypeInvalid) ...`.

Comment (DenisVieriu97) [reply aten/src/ATen/native/mps/OperationUtils.h:67]: Thanks - I didn't know there is a MPSDataTypeInvalid for MPSDataType

Comment (DenisVieriu97) [reply aten/src/ATen/native/mps/OperationUtils.h:67]: Updated

**Review by razarmehr — APPROVED** [review]
LGTM

**Pull request merged**